corehq/apps/userreports/expressions/json_query_specs.py

```diff
@@ -0,0 +1,98 @@
+import itertools
```

Comment:
nit: `json_query_specs` feels like a weird name to me. maybe `list_specs` or `items_specs` or something?